### ISSUE TITLE
Evolution of JupyterLab into a Cloud IDE

### ISSUE THREAD

**Issue by dclong** [jupyterlab/jupyterlab#3516]
I came across an interesting product which is RIDE Server for R, Python and SQL. From the appearance, I can tell that it leverages JupyterLab. However, it goes much further. It feels like a modern cloud IDE. 

https://r-brain.io/en/rideserver/

**Comment by jasongrout** [comment]
Yes, the r-brain folks and us have worked together on JupyterLab features too. They have been using JupyterLab for a while and we talk on occasion.

Was there something specific you wanted to discuss about this, or just to bring it to our attention?

**Comment by dclong** [comment]
I just want to bring this up to wider awareness. It's awesome to see nice products based JupyterLab to come out. 

In terms of features that JupyterLab miss badly to be an IDE, it would be debugging. There are already tickets on this.

**Comment by MMesch** [comment]
The triaging agent has identified this issue as resolved. The core request for debugging features making JupyterLab more IDE-like has been addressed. JupyterLab now includes comprehensive debugging capabilities introduced around January 2020, including breakpoints, variable inspection, step debugging, call stack navigation, and debug console support across notebooks, consoles, and file editors. Other IDE-features should be tracked in specific issues. Two related issues with further discussion on this topic are:

- #5234
- #4200
